File: backend/server/update.go
package server

const (
	MESSAGE = "Message"
	CHANNEL = "Channel"
	ROLE    = "Role"
	MEMBER  = "Member"
	DELETE  = "Delete"
	MODIFY  = "Modify"
	NEW     = "New"
)

type Update struct {
	Type    string  `json:"type"`
	Event   string  `json:"event"`
	Server  Server  `json:"server"`
	Channel Channel `json:"channel"`
	Role    Role    `json:"role"`
	Message Message `json:"message"`
}

File: frontend/src/app/shared/models/update.ts
import { Channel } from "./channel";
import { Message } from "./message";
import { Role, Server } from "./server";

export interface Update {
    type: UpdateType
    event: UpdateEvent
    server?: Server
    channel?: Channel
    role?: Role
    message?: Message
}

export enum UpdateEvent {
    MESSAGE = "Message",
    CHANNEL = "Channel",
    ROLE = "Role",
    MEMBER = "Member"
}

export enum UpdateType {
    DELETE = "Delete",
    MODIFY = "Modify",
    NEW = "New",
}
